### PR TITLE
First iteration 2D Transfer Functions

### PR DESCRIPTION
**Pull request by alvarosan**
First iteration of 2D transfer functions in tomviz.

Supersedes #1155 

tomviz_2dtf

### CODE REVIEW

File: tomviz/Module.h [hunk 1]
```diff
@@ -42,6 +43,16 @@ class Module : public QObject
...
   ~Module() override;
 
+  /// Transfer function mode (1D or 2D). This enum needs to be synchronized with
+  /// the order of the tabs in ui->tabWidget1D2DTransfer.
+  enum TransferMode
+  {
+    TF_1D = 0,
+    TF_2D
+  };
+  void setTransferMode(const int mode);
```
Comment (alvarosan) [inline tomviz/Module.h]: Good catch

**Comment by cjh1** [comment]
@alvarosan CI should have the correct version of ParaView now.

**Comment by alvarosan** [comment]
Thanks @cjh1 .   I pushed my latest fixes.

**Comment by alvarosan** [comment]
@cryos , should we merge this?

**Comment by mathturtle** [comment]
@alvarosan If I open tomviz, load in a dataset and exit tomviz it leaks a vtkImageData (According to vtkDebugLeaks).  `git bisect` points at this branch, but I don't want to recompile ParaView so I can't bisect further...

**Comment by cryos** [comment]
I think it would be best to report an issue and track it there rather than here (reference this PR). It is new functionality, with new API that was added to VTK. It would be good to see if @alvarosan can reproduce the leak and track it down, thanks for catching this.

**Comment by alvarosan** [comment]
Good catch @mathturtle , I created an issue for it.

**Pull request merged**